### ISSUE TITLE
Failing landIce_FO_GIS_AdjointSensitivity_Thickness test in nightly spack build

### ISSUE THREAD

**Issue by ikalash** [sandialabs/Albany#1054]
The ```landIce_FO_GIS_AdjointSensitivity_Thickness``` test has been failing in the nightly spack build.  This started a few weeks ago, but I did not get a chance to look at it until today due to travel.

The attachment contains output from the test.  Why is so much info about the MDFields dumped?  That makes the output very difficult to parse through.  Curiously, it looks like none of the regular nightlies are running this test, just the spack one.  Is there a particular configure option that triggers the test to be turned on?

@mperego @bartgol

**Comment by ikalash** [comment]
Thanks for looking at it, @bartgol .  Yes, it looks like ```ENABLE_MESH_DEPENDS_ON_PARAMETERS``` is only turned on in the Intel CEE build: https://github.com/sandialabs/Albany/blob/master/doc/dashboards/cee-compute011.sandia.gov/ctest_nightly_tmp.cmake#L489-L518 .  It's odd to me that it's not on in the others.  I can certainly turn it on in more nightlies - I will do this.  Let me know if there are any specific ones you would like it enabled in.  The Spack build does not push to the CDash site, so that is a separate thing.  I bet if the option is enabled, we will see the same failure as in the spack build in the nightlies.  We can see tomorrow.

Unfortunately the Intel build has been broken due to another issue, and right now the CDash site it down so I cannot check the history of this test in that build.  I will try again later today.

**Comment by bartgol** [comment]
The only drawback of that option is to make the build/run time longer. So I would maybe enable it on machines that are not already slow, to make sure we get a report in a reasonable time...

**Comment by bartgol** [comment]
Looking at the git history, there are two things that happened on the 23rd, which could affect testing:
- PR #1051 was merged
- you pushed a couple of commits to master, related to ceecompute dashboards.

It's hard to believe any of those had an impact on the pass/fail of the test. We can look at the test log of the last passing test, and compare against the current log, to see if any of the parameters has somehow changed. But given the date you pinpointed, I doubt we'll see a difference. Maybe something changed in Trilinos? Like a default value in Piro or some other solver package? @mperego ?

**Comment by ikalash** [comment]
Ok I can try to do bisection to figure out what broke things on 5/23.  Please stay tuned.

**Comment by ikalash** [comment]
It is the commits on 5/23 made by @mcarlson801 that are the culprit.  I checked out the following shas:

- From 5/23 00:00: 5fa2847f97ec3b10501d0b4d90fa3360c9ba1504 -> good
- From 5/24 00:00: 582623da51100ad0c8023fe72b31e414a631a6ee -> bad 

It is not from Trilinos - I fixed the Trilinos and still reproduced the issue.  @mcarlson801 : could you please look into this.

**Comment by mcarlson801** [comment]
Yeah, I'll look into this. I assume I can replicate this elsewhere by enabling `ENABLE_MESH_DEPENDS_ON_PARAMETERS`?

**Comment by ikalash** [comment]
@mcarlson801 : correct.  Thanks!

**Comment by mcarlson801** [comment]
I just pushed a change that should fix this. I had moved some temporary DynRankViews for the grounding line flux response into the kokkos kernel but this seemed to introduce some inaccuracy to the local responses which threw off the sensitivity computation when using certain scalar types.

**Comment by ikalash** [comment]
Thanks, @mcarlson801 !  I'll close this once the nightlies are clean.  Unfortunately Trilinos has broken the Albany nightlies and we may have to wait until that is fixed to check that things are clean.

**Comment by mcarlson801** [comment]
Looks like my fix for this might have broken some other tests using grounding line flux response. I'm probably going to just revert the uvm-free changes to the GLFlux evaluator and try to figure out what's going on.
